store/src/java/com/zimbra/cs/service/mail/FolderAction.java

```diff
@@ -167,6 +169,10 @@ private String handleFolder(Map<String,Object> context, Element request, String
             String zid = action.getAttribute(MailConstants.A_ZIMBRA_ID);
             mbox.revokeAccess(octxt, iid.getId(), zid);
         } else if (operation.equals(OP_GRANT)) {
+            if (!LC.zimbra_root_folder_sharing_enabled.booleanValue() && iid != null
```

Comment:
I think, so we should first check the folder type is root or not, then after should check the LC attribute.

Reply:
I would do it like.
```
protected boolean isRootFolder(iid) {
   return (iid.getId() == FolderConstants.ID_FOLDER_USER_ROOT || iid.getId() == FolderConstants.ID_FOLDER_ROOT);
}

if (!isRootFolder(iid) && !LC.zimbra_root_folder_sharing_enabled.booleanValue()) {

}
```